CMakeLists.txt

```diff
@@ -115,27 +118,49 @@ else()
 endif()
 
 add_library(pmemkv SHARED ${SOURCE_FILES})
-target_link_libraries(pmemkv ${LIBPMEMOBJ++_LIBRARIES})
-target_link_libraries(pmemkv ${MEMKIND_LIBRARIES})
-target_link_libraries(pmemkv ${RapidJSON_LIBRARIES})
+
+target_link_libraries(pmemkv PRIVATE ${LIBPMEMOBJ++_LIBRARIES}
+                                    ${MEMKIND_LIBRARIES}
+                                    ${RapidJSON_LIBRARIES}
+                                    ${TBB_LIBRARIES})
 
 target_include_directories(pmemkv PRIVATE src/valgrind)
 
 # Enable libpmemobj-cpp valgrind annotations
 target_compile_options(pmemkv PRIVATE -DLIBPMEMOBJ_CPP_VG_ENABLED=1)
 
-if(TBB_IMPORTED_TARGETS)
-    target_link_libraries(pmemkv ${TBB_IMPORTED_TARGETS})
-else()
-    target_link_libraries(pmemkv tbb)
-endif(TBB_IMPORTED_TARGETS)
-
 if(EXPERIMENTAL)
-    target_link_libraries(pmemkv ${CMAKE_THREAD_LIBS_INIT} memcached)
-    target_link_libraries(pmemkv acl_cpp protocol acl)
+    target_link_libraries(pmemkv PRIVATE ${CMAKE_THREAD_LIBS_INIT} memcached)
+    target_link_libraries(pmemkv PRIVATE acl_cpp protocol acl)
 endif(EXPERIMENTAL)
 
+set_target_properties(pmemkv PROPERTIES PUBLIC_HEADER src/libpmemkv.h)
+
+# The effect of the below is that when user program links with pmemkv target
+# it's include_directories are automatically set to CMAKE_INSTALL_PREFIX/include
+# The user program does not have to call include_directories() manually.
+target_include_directories(pmemkv
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/src)
+
+install(TARGETS pmemkv EXPORT libpmemkv-config
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+install(EXPORT libpmemkv-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libpmemkv)
```

Comment:
I believe it's wrong to export two ways of discovering pmemkv - one for cmake projects and another for everything else. It will require more tests to verify that both ways work.